src/utils/helpers/GraphHelpers.ts

```diff
@@ -86,36 +71,149 @@ export function selectHighLightedNodes() {
 }
 
 export function selectAllNodes() {
-    return select('#labels').selectAll<SVGGElement, DependencyNode>('g');
+    return select(Selectors.LABELS).selectAll<SVGGElement, DependencyNode>('g');
+}
+
+export function selectHighlightBackground() {
+    return select(Selectors.HIGHLIGHT_BACKGROUND);
+}
+
+function selectDetailsButtonWrapper() {
+    return select(Selectors.DETAILS_BUTTON);
+}
+
+export function selectDetailsButtonRect() {
+    return selectDetailsButtonWrapper().select('rect');
+}
+
+export function selectDetailsButtonText() {
+    return selectDetailsButtonWrapper().select('text');
 }
 
 export function centerScreenToDimension(dimension: ReturnType<typeof findGroupBackgroundDimension>, scale?: number) {
     if (dimension) {
-        const svgContainer = select('#container');
+        const svgContainer = select(Selectors.CONTAINER);
 
         const width = Number(svgContainer.attr('width'));
         const height = Number(svgContainer.attr('height'));
 
-        const scaleValue = Math.min(1.3, 0.9 / Math.max(dimension.width / width, dimension.height / height));
+        const scaleValue = scale || Math.min(1.3, 0.9 / Math.max(dimension.width / width, dimension.height / height));
 
         svgContainer
+            .attr('data-scale', scaleValue)
```

Comment:
does it have to be stored in attribute?

Reply:
It can either be stored in attribute or global variable, or wrapped in a basic class. I decided to store it as an attribute in a container because it is strictly related to graph